conda/env/env.py

```diff
@@ -207,7 +219,400 @@ def add(self, package_name):
+        }
+
+        :return: A list of conda or PyPI package specs
+        """
+        result = []
+        for req in self:
+            result.append(req.serialize())
+
+        return result
+
+    @classmethod
+    def from_strs(
+        cls,
+        raw_requirements: Iterable[str | dict[str, str]] | None = None,
+        raw_pypi_requirements: Iterable[str | dict[str, str]] | None = None,
+    ) -> Requirements:
+        """Initialize a set of requirements from lists of string conda and PyPI requirements.
+
+        {
+            "if": "__win",
+            "then": "flask"
+        }
+
+        :param raw_requirements: A list of string conda requirements
+        :param raw_pypi_requirements: A list of string PyPI requirements
+        :return: A new Requirements instance containing the requested package specs
+        """
+        requirements: list[Requirement] = []
+        pypi_requirements: list[Requirement] = []
+
+        if raw_requirements is None:
+            raw_requirements = []
+
+        if raw_pypi_requirements is None:
+            raw_pypi_requirements = []
+
+        for raw_req in raw_requirements:
+            requirements.append(Requirement(raw_req))
+
+        for raw_req in raw_pypi_requirements:
+            pypi_requirements.append(Requirement(raw_req))
+
+        return cls(requirements=requirements, pypi_requirements=pypi_requirements)
+
+    def __iter__(self) -> Iterator[Requirement]:
+        yield from self.requirements
+        yield from self.pypi_requirements
+
+
+class EnvironmentConfig:
+    """A class that contains configuration variables for an environment.
+
+    These variables override user configuration, e.g. settings in .condarc.
+    """
+
+    def __init__(self, **options):
+        self.options = options
+
+    def __repr__(self) -> str:
+        lines = []
+        for key, value in self.options.items():
+            lines.append(f"{key}: {value}")
+        return "\n".join(lines)
+
+    def serialize(self) -> dict[str, str]:
+        """Serialize the environment configuration.
+
+        Since this is a thin wrapper around a dictionary, just return the underlying
+        dict.
+
+        :return: A dict of primitives containing configuration options for the
+            environment
+        """
+        return self.options
+
+
+class EnvironmentBase:
+    """A base class representing an ``environment.yaml`` file"""
+
+    def to_dict(self) -> dict:
+        raise NotImplementedError
+
+    def to_yaml(self) -> str | None:
+        raise NotImplementedError
+
+    def save(self):
+        raise NotImplementedError
+
+
+class EnvironmentV2(EnvironmentBase):
+    """A class representing a V2 environment.yaml."""
+
+    def __init__(
+        self,
+        requirements: Requirements | None = None,
+        groups: dict[str, Requirements] | None = None,
+        name: str | None = None,
+        config: EnvironmentConfig | None = None,
+        **options,
+    ):
```

Comment:
Where's the description?

Reply:
Ack, I can add a description now - usually I don't for `__init__` because it's always the same: "Initialize an instance of <whatever class>". But here I think having annotations for e.g. `options` would be helpful.